packages/bitcore-wallet-client/src/lib/common/utils.ts

```diff
@@ -511,6 +513,29 @@ export class Utils {
           gasLimit
         };
         unsignedTxs.push(Transactions.create({ ...txp, ...multiSendParams }));
+      } else if (multiTx) {
+        // Add unsigned transactions in outputOrder
+        for (let index = 0; index < outputOrder.length; index++) {
+          const outputIdx = outputOrder[index];
+          if (!outputs?.[outputIdx]) {
+            throw new Error('Output index out of range');
+          }
+          const recepient = {
+            amount: outputs[outputIdx].amount,
+            address: outputs[outputIdx].toAddress,
+            tag: outputs[outputIdx].tag
+          }
+          const _tag = recepient?.tag || destinationTag;
+          const rawTx = Transactions.create({
+            ...txp,
+            ...recepient,
+            tag: _tag ? Number(_tag) : undefined,
+            chain: _chain,
+            nonce: Number(txp.nonce) + Number(index),
```

Comment:
:thinking: This is some hidden logic. So the txp.nonce isn't actually the nonce that gets sent (except the first tx). I guess this is ok in this XRP case... An alternative would be to put the nonces/sequences explicitly on the individual outputs objects, but either way there's going to need to be some special logic around finding the next nonce when "stacking" txps. Something to be aware.

Reply:
Right. All the info is there to be able to determine what the sequence is for each output. I'm just not sure if it's worth being explicit in order to avoid confusion if something goes wrong, or if it hinders extensibility. I'm inclined to say what you have is fine...again the only real point of confusion that I foresee is when trying to determine the next nonce when stacking txps. I would not expect fresh eyes to the system to guess that the next nonce would need to be `txp.nonce + Math.max(txp.outputOrder)`. The worst case, though, is that the nonces overlap and some txs fail to send...which explicitly setting the nonce on the outputs wouldn't prevent, but just potentially reduce the likelihood.

Anyway, I'm not suggesting that one way or the other is better. Just trying to verbally process between the two for your consideration. I'll let you decide which to go with.